backend/src/test/kotlin/org/darkSolace/muse/securityModule/AuthControllerApiTests.kt

```diff
@@ -0,0 +1,188 @@
+            url,
+            SignUpRequest("test", "123", "test2@example.com"),
+            String::class.java
+        )
+
+        Assertions.assertEquals(HttpStatus.BAD_REQUEST, secondResponse.statusCode)
+        Assertions.assertEquals("Error: Username is already in use!", secondResponse.body)
+    }
+
+    @Test
+    @Order(3)
+    fun testSignUp_DuplicateEMail() {
+        val url = generateUrl("/api/auth/signup")
+        val response = restTemplate.postForEntity(
+            url,
+            SignUpRequest("test", "123", "test@example.com"),
+            String::class.java
+        )
+        Assertions.assertEquals(HttpStatus.OK, response.statusCode)
+        Assertions.assertEquals("User created successfully.", response.body)
+
+        val secondResponse = restTemplate.postForEntity(
+            url,
+            SignUpRequest("test2", "123", "test@example.com"),
+            String::class.java
+        )
+        Assertions.assertEquals(HttpStatus.BAD_REQUEST, secondResponse.statusCode)
+        Assertions.assertEquals("Error: Email is already in use!", secondResponse.body)
+    }
+
+    @Test
+    @Order(4)
+    fun testSignIn() {
+        val url = generateUrl("/api/auth/signup")
+        val response = restTemplate.postForEntity(
+            url,
+            SignUpRequest("test", "123", "test@example.com"),
+            String::class.java
+        )
+        Assertions.assertEquals(HttpStatus.OK, response.statusCode)
+        Assertions.assertEquals("User created successfully.", response.body)
+
+        val url2 = generateUrl("/api/auth/signin")
+        val secondResponse = restTemplate.postForEntity(
+            url2,
+            SignUpRequest("test", "123", "test@example.com"),
+            JwtResponse::class.java
+        )
+        Assertions.assertEquals(HttpStatus.OK, secondResponse.statusCode)
+        Assertions.assertNotNull(secondResponse.body)
+        Assertions.assertInstanceOf(JwtResponse::class.java, secondResponse.body)
+        Assertions.assertEquals("test", secondResponse.body?.username)
+    }
+
+    @Test
+    @Order(5)
+    fun testSignIn_WrongPassword() {
+        val url = generateUrl("/api/auth/signup")
+        val response = restTemplate.postForEntity(
+            url,
+            SignUpRequest("test", "123", "test@example.com"),
+            String::class.java
+        )
+        Assertions.assertEquals(HttpStatus.OK, response.statusCode)
+        Assertions.assertEquals("User created successfully.", response.body)
+
+        val url2 = generateUrl("/api/auth/signin")
+        val secondResponse = restTemplate.postForEntity(
+            url2,
+            SignUpRequest("test", "1234", "test@example.com"),
+            String::class.java
+        )
+        Assertions.assertEquals(HttpStatus.UNAUTHORIZED, secondResponse.statusCode)
+        Assertions.assertEquals("Username or password wrong!", secondResponse.body)
+
+    }
+
+    @Test
+    @Order(6)
+    fun testSignIn_SuspendedUser() {
```

Comment:
Can't find the right place in the code, please add a TODO both here in the test and where ever the unauthorized status code is selected that suspended users may still login, they just aren't allowed to do anything until they confirm their warning and get the suspended role removed.

Reply:
is related to the previous open comment comment (re: 4 functions return the same). Waiting for clarification.